public/app/plugins/datasource/cloudwatch/components/ui/EditorList.tsx

```diff
@@ -0,0 +1,46 @@
+import { Button } from '@grafana/ui';
+import React from 'react';
+import Stack from './Stack';
+
+interface EditorListProps<T> {
+  items: T[];
+  renderItem: (item: Partial<T>, onChangeItem: (item: T) => void, onDeleteItem: () => void) => React.ReactElement;
+  onChange: (items: Array<Partial<T>>) => void;
+}
+
+function EditorList<T>({ items, renderItem, onChange }: EditorListProps<T>) {
+  const onAddItem = () => {
+    const newItems = [...items, {}];
+
+    onChange(newItems);
+  };
+
+  const onChangeItem = (itemIndex: number, newItem: T) => {
+    const newItems = [...items];
+    newItems[itemIndex] = newItem;
+    onChange(newItems);
+  };
+
+  const onDeleteItem = (itemIndex: number) => {
+    const newItems = [...items];
+    newItems.splice(itemIndex, 1);
+    onChange(newItems);
+  };
+
+  return (
+    <Stack>
+      {items.map((item, index) => (
+        <div key={index}>
```

Comment:
I know I made this 😂  - but looking at it again again if possible, i think `<Fragment key={index}`/>` would be cleaner, if possible.

There's the possibility this causes regressions in the few places its used, so I wouldn't stress too much about it.